ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -378,7 +378,7 @@ public void close() {
         referenceTimeInMs);
     long startTimeMs = time.milliseconds();
     Map<StoreKey, Long> deletedKeys = new HashMap<>();
-    NavigableMap<String, Long> validSizePerLogSegment = new TreeMap<>();
+    NavigableMap<String, Long> validSizePerLogSegment = new TreeMap<>(LogSegmentNameHelper.COMPARATOR);
```

Comment:
can you create an issue to add tests in BlobStoreStatsTest to test this scenario (I assume you haven't covered in this patch)